Using pytket backend for authentication calls

chris-theroux opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ X] I have added the tests to cover my changes.
- [ X] I have updated the documentation accordingly.
- [ X] I have read the CONTRIBUTING document.
-->

### Summary
Using the Quantinuum Pytket library for authentication calls.
Still making use of Qiskit credential storage for tokens / user information.


### Details and comments
This is the first of a few major changes we are making.
The goal is to replace all API calls with the Pytket implementation.
Please let me know if any of these changes are in conflict with Qiskit's design.

chris-theroux committed:
Using pytket backend for authentication calls

chris-theroux commented:
Working on fixing py3.6 build error.

chris-theroux commented:
@mtreinish is this an error on my side? 
Version 3.[6](https://github.com/qiskit-community/qiskit-quantinuum-provider/actions/runs/3585155799/jobs/6032764037#step:3:7) was not found in the local cache
Error: Version 3.6 with arch x64 not found
The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json

chris-theroux committed:
Removing py3.6 and updating QuantinuumExample

chris-theroux commented:
Removed python 3.6 support and updated the examples notebook.

mtreinish commented:
I apologize, I was on vacation at the end of the year when you opened this and it completely slipped through the cracks. I took a quick look and the failures are primarily because of the tox 4.0.0 package release (which was incompatible with earlier version). Let me push up a Pr to get CI happy and then I'll rebase this for you.

mtreinish committed:
Merge branch 'master' into pytket-login

mtreinish commented on qiskit_quantinuum/VERSION.txt:
```diff
@@ -1 +1 @@
-0.3.0
\ No newline at end of file
+0.3.1
```
I think we should probably use 0.4.0 since this is the first release with the new package name:

```suggestion
0.4.0
```

mtreinish committed:
Update version to be 0.4.0

mtreinish commented:
Looks like we'll have to drop 3.7 support too: https://github.com/qiskit-community/qiskit-quantinuum-provider/actions/runs/4564705714/jobs/8054817824#step:6:23 it looks like `pytket-quantinuum>=0.9`  is only Python>=3.8. Let me make that change locally. Assuming CI passes with that change made this LGTM.

mtreinish committed:
Raise minimum supported python version to 3.8

The minimum supported version of our dependencies after this PR is
Python 3.8. This will require us to drop 3.7 support completely. This is
probably reasonable because this will be the first release of the new
package and also Python 3.7 is slated to go EoL this summer, so dropping
3.7 support now is reasonable as that version of Python is sunset.

mtreinish reviewed (approved):
Overall LGTM. The only question I have before merging is I noticed in the CI logs there are a bunch of warning messages like:

```
Error: URL http://testapi.quantinuum.com/v1 didn't conform to expected form.
Using default url and version: https://qapi.quantinuum.com/ and v1.
```

https://github.com/qiskit-community/qiskit-quantinuum-provider/actions/runs/4564798681/jobs/8055023115#step:6:25

Is that a real issue or just something that's popping up in the tests?

mtreinish commented:
I went ahead and merged this, we can fix the tests later if needed.

chris-theroux commented:
Hey @mtreinish, getting back to this project as well. Going to take a look at the test failures and double check that login is still working w/ the new version
